invenio_records_resources/records/systemfields/files/manager.py

```diff
@@ -241,13 +241,27 @@ def delete(self, key, remove_obj=True, softdelete_obj=True, remove_rf=False):
     @ensure_enabled
     def delete_all(self, remove_obj=True, softdelete_obj=True, remove_rf=False):
         """Delete all file records."""
+        object_versions = []
         for key in list(self.keys()):
-            self.delete(
-                key,
-                remove_obj=remove_obj,
-                softdelete_obj=softdelete_obj,
-                remove_rf=remove_rf,
-            )
+            rf = self[key]
+            if rf.object_version:
+                object_versions.append(rf.object_version)
+            # Remove or softdelete the entire row
+            rf.delete(force=remove_rf)
+
+        if remove_obj:
+            if softdelete_obj:
+                for object_version in object_versions:
+                    ObjectVersion.delete(object_version.bucket, object_version.key)
```

Comment:
do I understand correctly that soft delete is still handled on one by one basis ?